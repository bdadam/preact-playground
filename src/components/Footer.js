/** @jsx h */

import './Footer.scss';

import { h, Component } from 'preact';

export default class Footer extends Component {
    shouldComponentUpdate() { return false; }

    render(props, state) {
        return (
            <footer class="site-footer">
                &copy; 2017
                &nbsp;
                <a href="/">Legal notes</a>
            </footer>
        );
    }
}