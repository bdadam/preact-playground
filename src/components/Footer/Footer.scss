.site-footer {
    padding: 20px;
    background-color: #f1f1f1;
    border-top: 1px solid #cdcdcd;
    text-align: center;
}
