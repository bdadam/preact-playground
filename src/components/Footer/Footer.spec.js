import { h } from 'preact';
import render from 'preact-render-to-string';

import Footer from './Footer';

describe('Footer', () => {

    it('has link to legal notes', () => {
        const footer = render(<Footer/>);
        const linkToLegalNotes = render(<a href="/">Legal notes</a>);
        expect(footer).toContain(linkToLegalNotes);
    });

    it('matches snapshot', () => {
        const footerTree = render(<Footer/>);
        expect(footerTree).toMatchSnapshot();
    });
});