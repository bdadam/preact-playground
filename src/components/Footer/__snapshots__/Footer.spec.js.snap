// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<footer class="site-footer">© 2017  
  <a href="/">Legal notes</a>
</footer>
`;
